src/sys/statfs.rs

```diff
@@ -1,22 +1,120 @@
-use {Result, NixPath};
+use {NixPath, Result};
 use errno::Errno;
+use libc::{self, c_ulong};
+use std::mem;
 use std::os::unix::io::AsRawFd;
-use libc;
 
-pub fn statfs<P: ?Sized + NixPath>(path: &P, stat: &mut libc::statfs) -> Result<()> {
-    unsafe {
-        Errno::clear();
-        let res = try!(
-            path.with_nix_path(|path| libc::statfs(path.as_ptr(), stat))
-        );
+#[cfg(target_os = "android")]
+pub type fsid_t = libc::__fsid_t;
+#[cfg(not(target_os = "android"))]
+use libc::fsid_t;
+
+#[derive(Clone, Copy)]
+#[allow(missing_debug_implementations)]
+pub struct Statfs(libc::statfs);
+
+impl Statfs {
+    /// Magic code defining system type
+    pub fn filesystem_type(&self) -> c_ulong {
+        self.0.f_type as c_ulong
```

Comment:
This API is really nasty.  Depending on the platform, `f_type` can be defined as either a long, ulong, int, uint32_t, or uint64_t.  On Linux, you'll most likely want to compare it to the magic constants, which are defined as either long or int.  On BSD, the value means nothing by itself, and there are no constants to compare it to.  So I think the best API that Nix can present would be to return a Newtype that derives `Eq`.  On Linux, it can define constants for the various magic constants.  On BSD, it should define none, and be totally opaque.

Reply:
Yeah, but use the constants from libc instead of hard-coding them.  And the inner type should be whatever libc exports instead of u64.